network/p2p/p2p_test.go

```diff
@@ -225,49 +257,86 @@ func TestWaitSubsetOfPeers(t *testing.T) {
 	}
 }
 
-func dummyMsgCommittee(t *testing.T, pkHex string, height int) (spectypes.MessageID, *spectypes.SignedSSVMessage) {
-	return dummyMsg(t, pkHex, height, spectypes.RoleCommittee)
+func generateMsg(ks *spectestingutils.TestKeySet, round specqbft.Round) *spectypes.SignedSSVMessage {
+	netCfg := networkconfig.TestNetwork
+	height := specqbft.Height(netCfg.Beacon.EstimatedCurrentSlot())
+
+	share := &ssvtypes.SSVShare{
+		Share: *spectestingutils.TestingShare(ks, spectestingutils.TestingValidatorIndex),
+		Metadata: ssvtypes.Metadata{
+			BeaconMetadata: &beaconprotocol.ValidatorMetadata{
+				Status: eth2apiv1.ValidatorStateActiveOngoing,
+				Index:  spectestingutils.TestingShare(ks, spectestingutils.TestingValidatorIndex).ValidatorIndex,
+			},
+			Liquidated: false,
+		},
+	}
+	committeeID := share.CommitteeID()
+
+	fullData := spectestingutils.TestingQBFTFullData
+
+	encodedCommitteeID := append(bytes.Repeat([]byte{0}, 16), committeeID[:]...)
+	committeeIdentifier := spectypes.NewMsgID(netCfg.DomainType(), encodedCommitteeID, spectypes.RoleCommittee)
```

Comment:
this function only generate committee role message, we should add other single validator runners to this test messages as well. 

task: add validator duties (proposer/synccommitteecontribution/aggregator) to generatemsg or create another function and use in `TestP2pNetwork_SubscribeBroadcast`